File: joey/week3-220728/2011. Final Value of Variable After Performing Operations.py
"""
Runtime: 49 ms, faster than 97.09% of Python3 online submissions for Final Value of Variable After Performing Operations.
Memory Usage: 13.8 MB, less than 54.80% of Python3 online submissions for Final Value of Variable After Performing Operations.
"""
class Solution:
    def finalValueAfterOperations(self, operations: List[str]) -> int:
        res = 0
        for operation in operations:
            res += 1 if '+' in operation else -1
        return res

File: billy/week3-220728/2011. Final Value of Variable After Performing Operations.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var finalValueAfterOperations = function(operations) {
    let result = 0;

    for (let i = 0; i < operations.length; i += 1) {
        const operation = operations[i];

        if (operation === 'X++' || operation === '++X') {
            result += 1;
        } else if (operation === 'X--' || operation === '--X') {
            result -= 1;
        }
    }

    return result;
};